src/middlewared/middlewared/plugins/jail.py

```diff
@@ -106,18 +106,27 @@ def create_job(self, job, options):
     @accepts(Str("jail"), Dict(
              "options",
              Bool("plugin"),
+             Bool("rename"),
              additional_attrs=True,
              ))
     def do_update(self, jail, options):
         """Sets a jail property."""
         plugin = options.pop("plugin")
+        rename = options.pop("rename")
+
         iocage = ioc.IOCage(skip_jails=True, jail=jail)
 
+        if rename:
+            new_name = options.pop("name")
```

Comment:
That way if `rename` is set but `name` is not provided you will have a traceback.

Solution is to check name is set, raising a ValidationError or just check for `name`.
I dont think one need to set `rename` if already providing a `name`.